py/examples/table_pagination_pandas.py

```diff
@@ -0,0 +1,98 @@
+# Table / Pagination / Pandas
+# Use a paginated #table to display large (100k+ rows) tabular data using pandas dataframe.
+# #form #table #pagination #pandas
+# ---
+
+from h2o_wave import main, app, Q, ui
+import pandas as pd
+
+
+all_issues_df = pd.DataFrame(
+    [[i + 1, 'Closed' if i % 2 == 0 else 'Open'] for i in range(100)],
+    columns=['text', 'status']
+)
+rows_per_page = 10
+total_rows = len(all_issues_df)
+
+
+def get_table_rows(q: Q):
+    # Make a deep copy in order to not mutate the original df which serves as our baseline.
+    df = all_issues_df.copy()
+
+    if q.client.sort:
+        # Reverse values since default sort of Wave table is different from Pandas.
+        ascending = [not v for v in list(q.client.sort.values())]
+        df = df.sort_values(by=list(q.client.sort.keys()), ascending=ascending)
+    # Filter out all rows that do not contain searched string in `text` cell.
+    if q.client.search:
+        df = df[df.text.apply(str).str.contains(q.client.search)]
+    # Filter out rows that do not contain filtered column value.
+    if q.client.filters:
+        # We want only rows that have no filters applied or their col value matches active filters.
+        query = ' & '.join([f'({not bool(filters)} | {col} in {filters})' for col, filters in q.client.filters.items()])
+        df = df.query(query)
+
+    # Update table pagination according to the new row count.
+    if q.client.search is not None or q.client.filters:
+        q.page['form'].items[0].table.pagination = ui.table_pagination(len(df), rows_per_page)
+
+    offset = q.client.page_offset or 0
+    df = df[offset:offset + rows_per_page]
+    return [ui.table_row(name=str(r[0]), cells=[str(r[0]), r[1]]) for r in df.itertuples(index=False)]
+
+
+@app('/demo')
+async def serve(q: Q):
+    if not q.app.initialized:
+        # Create and upload a CSV file for downloads.
+        all_issues_df.to_csv('data_download.csv')
+        q.app.data_download, = await q.site.upload(['data_download.csv'])
+        q.app.initialized = True
+
+    if not q.client.initialized:
+        q.page['meta'] = ui.meta_card(box='')
+        q.page['form'] = ui.form_card(box='1 1 -1 -1', items=[
+            ui.table(
+                name='table',
+                columns=[
+                    ui.table_column(name='text', label='Text', sortable=True, searchable=True),
```

Comment:
In the example let's make `link=False` so this column is not clickable OR handle clicking the row in the app

Reply:
Agree, done.

Reply:
Agree, done.